core/types/types.cc

```diff
@@ -663,8 +663,11 @@ SelfTypeParam::SelfTypeParam(const SymbolRef definition) : definition(definition
 }
 
 bool LambdaParam::derivesFrom(const GlobalState &gs, SymbolRef klass) const {
-    Exception::raise(
-        "LambdaParam::derivesFrom not implemented, not clear what it should do. Let's see this fire first.");
+    // This only ever gets called in bad situations when we haven't instantiated a LambdaParam as either a SelfTypeParam
+    // (within the body of a generic class) or a specific type (after having instantiated the class with specific
+    // parameters and/or T.untyped.) However, those errors are captured elsewhere, so here we can follow the same
+    // behavior as SelfTypeParam and return 'false'.
```

Comment:
`SelfTypeParam` is a valid type inside method bodies, while method bodies should _never_ see uninstantiated `LambdaParam`. It's like a free type variable, if you have it, the expression isn't well formed.